src/portal/mod.rs

```diff
@@ -218,6 +238,48 @@ impl Keyring {
     }
 }
 
+pub trait KeyExt {
+    fn get<'a>(&'a self) -> Pin<Box<dyn std::future::Future<Output = &crate::Key> + 'a>>;
+}
+
+impl KeyExt for Keyring {
+    fn get<'a>(&'a self) -> Pin<Box<dyn std::future::Future<Output = &crate::Key> + 'a>> {
+        Box::pin(async {
+            if let Some(key) = self.key.get() {
+                key
+            } else {
+                let keyring = self.keyring.lock().await;
+                self.key.get_or_init(|| keyring.derive_key(&self.secret))
+            }
+        })
+    }
+}
+
+impl KeyExt for &Keyring {
```

Comment:
Do we need to provide an impl for `Keyring` & `&Keyring`? As this is only used internally, I guess not?